File: Selenium/Project/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver import ActionChains
import time

with webdriver.Firefox() as driver:
    driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login")
    wait = WebDriverWait(driver, timeout=10)
    driver.implicitly_wait(8)

    username_field = driver.find_element(By.XPATH,"//input[contains(@name,'Username')]")
    password_field = driver.find_element(By.XPATH,"//input[contains(@name,'Password')]")
    login_button = driver.find_element(By.XPATH,"//input[@name='Submit']")
    
    username_field.send_keys("orange")
    password_field.send_keys("orangepassword123")
    login_button.click()

    pimoption = driver.find_element(By.XPATH,"//a[contains(@id,'menu_pim_viewPimModule')]/b")
    pimoption.click()
    
    Add_button = wait.until(EC.visibility_of_element_located((By.XPATH,"//a[contains(@id,'addEmployee')]")))
    Add_button.click()

    firstName_field = driver.find_element(By.XPATH,"//input[@id='firstName']")
    lastName_field = driver.find_element(By.XPATH,"//input[@id='lastName']")
    save_button = driver.find_element(By.XPATH,"//input[@id='btnSave']")

    firstName_field.send_keys("Damon")
    lastName_field.send_keys("Salvatore")
    save_button.click()
    print('save button is clicked')

    Pim_option = wait.until(EC.element_to_be_clickable((By.XPATH,"//a[contains(@id,'menu_pim_viewPimModule')]")))
    Pim_option.click()
    time.sleep(10)
    emp_name_input_search = driver.find_element(By.XPATH,"//input[@id='empsearch_employee_name_empName']")
    emp_name_input_search.send_keys("Damon Salvatore")
    

    search_button = driver.find_element(By.XPATH,"//input[@id='searchBtn']")
    search_button.click()
    firstName_entry = driver.find_element(By.XPATH,"//table[@id='resultTable']/tbody/tr[1]/td[3]").text
    print(firstName_entry)
    assert firstName_entry == "Damon"
    lastName_entry = driver.find_element(By.XPATH,"//table[@id='resultTable']/tbody/tr[1]/td[4]").text
    print(lastName_entry)
    assert lastName_entry == "Salvatore"

    id = driver.find_element(By.XPATH,"//table[@id='resultTable']/tbody/tr[1]/td[2]").text
    assert id != 'null'

    driver.quit()
    

File: Selenium/Project/Activity4.java
package project;
import static org.testng.Assert.assertEquals;

import java.time.Duration;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import net.bytebuddy.description.modifier.Visibility;



public class Activity4 
{
	WebDriver driver ;
	WebDriverWait wait;
	 @BeforeClass
	    public void beforeClass() {
	        driver = new FirefoxDriver();
	        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	        driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");
	    }
	
	 @Test
	    public void test() throws InterruptedException {
		
 	    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(6));

	        WebElement username = driver.findElement(By.xpath("//input[contains(@name,'Username')]"));
	        WebElement password = driver.findElement(By.xpath("//input[contains(@name,'Password')]"));
	        WebElement login = driver.findElement(By.xpath("//input[@name='Submit']"));
	        
	        username.sendKeys("orange");
	        password.sendKeys("orangepassword123");
	        login.click();
	        Thread.sleep(10);
	        
	        Actions builder = new Actions(driver);
    	    builder.moveToElement(driver.findElement(By.xpath("//a[contains(@id,'menu_pim_viewPimModule')]/b"))).click();
    	    
    	    Thread.sleep(5);
    	    WebElement Add_button = wait.until(ExpectedConditions.visibilityOfElementLocated((By.xpath("//a[contains(@id,'addEmployee')]"))));
    	    Add_button.click();

    	    WebElement firstName = driver.findElement(By.xpath("//input[@id='firstName']"));
    	    WebElement lastName = driver.findElement(By.xpath("//input[@id='lastName']"));
    	    WebElement save_button = driver.findElement(By.xpath("//input[@id='btnSave']"));

    	    firstName.sendKeys("Deepthi");
    	    lastName.sendKeys("Grandhi");
    	    save_button.click();
    	    System.out.println("save button is clicked");

    	    WebElement Pim_option = wait.until(ExpectedConditions.elementToBeClickable((By.xpath("//a[contains(@id,'menu_pim_viewPimModule')]"))));
    	    Pim_option.click();
    	    
    	    WebElement emp_name_input_search = driver.findElement(By.xpath("//input[@id='empsearch_employee_name_empName']"));
    	    emp_name_input_search.clear();
    	    emp_name_input_search.sendKeys("Deepthi Grandhi");
    	    

    	    WebElement search_button = driver.findElement(By.xpath("//input[@id='searchBtn']"));
    	    search_button.click();
    	    String firstName_entry = driver.findElement(By.xpath("//table[@id='resultTable']/tbody/tr[1]/td[3]")).getText();
    	    System.out.println(firstName_entry);
    	    Assert.assertEquals(firstName_entry, "Deepthi"); 
    	    String lastName_entry = driver.findElement(By.xpath("//table[@id='resultTable']/tbody/tr[1]/td[4]")).getText();
    	    System.out.println(lastName_entry);
    	    Assert.assertEquals(lastName_entry, "Grandhi"); 
    	 
    	    String id = driver.findElement(By.xpath("//table[@id='resultTable']/tbody/tr[1]/td[2]")).getText();
    	    Assert.assertNotNull(id);
    	
    }

	    @AfterClass
	    public void tearDown() {
	        driver.quit();
	    }


}
